home/utils.py

```diff
@@ -631,3 +432,284 @@ def export_csv_content(queryset: PageQuerySet, response: HttpResponse) -> None:
+            self.translation_tag = str(page.translation_key)
+            self.tags = self.__format_list_from_query_set(page.tags.all())
+            self.quick_replies = self.__format_list_from_query_set(
+                page.quick_replies.all()
+            )
+            self.triggers = self.__format_list_from_query_set(page.triggers.all())
+            self.locale = str(page.locale)
+            self.related_pages = self.__format_list_from_query_set(
+                self.__get_related_pages(page)
+            )
+
+    def get_row(self):
+        """Returns object as an exportable row"""
+        return [
+            self.side_panel_column_1,
+            self.side_panel_column_2,
+            self.side_panel_column_3,
+            self.side_panel_column_4,
+            self.side_panel_column_5,
+            self.side_panel_message_number,
+            self.parent,
+            self.web_title,
+            self.web_subtitle,
+            self.web_body,
+            self.whatsapp_title,
+            self.whatsapp_body,
+            self.messenger_title,
+            self.messenger_body,
+            self.viber_title,
+            self.viber_body,
+            self.translation_tag,
+            self.tags,
+            self.quick_replies,
+            self.triggers,
+            self.locale,
+            self.next_prompt,
+            self.image_link,
+            self.doc_link,
+            self.media_link,
+            self.related_pages,
+        ]
+
+    def __format_list_from_query_set(self, unformatted_query: PageQuerySet) -> str:
+        list_delimiter = ", "
+        return list_delimiter.join(str(x) for x in unformatted_query if str(x) != "")
+
+    def __get_parent_page(self, page: ContentPage) -> str:
+        """Get parent page title as string"""
+        if not HomePage.objects.filter(id=page.get_parent().id).exists():
+            return page.get_parent().title
+
+    def __get_messages(self, platform_body: blocks.StreamValue) -> List[dict]:
+        """Formats the platform body objects into a list of dictionaries"""
+        msgs = []
+        for msg in platform_body:
+            msgs.append(
+                {
+                    "msg": msg.value["message"].strip()
+                    if "message" in msg.value
+                    else None,
+                    "img": msg.value["image"] if "image" in msg.value else None,
+                    "doc": msg.value["document"] if "document" in msg.value else None,
+                    "med": msg.value["media"] if "media" in msg.value else None,
+                    "next_prompt": msg.value["next_prompt"]
+                    if "next_prompt" in msg.value
+                    else None,
+                }
+            )
+        return msgs
+
+    def __set_structure(self, structure_string: str):
+        """Sets the sidebar Menu/Sub level based on structure string length,
+        for example, Sub 1.2.1 is a level 3 message"""
+        if "menu" in structure_string.lower():
```

Comment:
will all index page shave menu in it? can we not find the depth from something like if it is a child of a homepage, then level is one, and so on

Reply:
the `structure_string` I passed through at the point `for index, page in enumerate(main_menu_pages, 1):` contains the string menu, the assumption I made from the template sheet I was sent that they only want "Menu" for the children of the homepage? but I might have misinterpreted the template